File: doc-examples/example-kotlin/src/main/kotlin/io/micronaut/gcp/pubsub/quickstart/AnimalClient.kt
package io.micronaut.gcp.pubsub.quickstart

//tag imports[]
import io.micronaut.gcp.pubsub.annotation.PubSubClient
import io.micronaut.gcp.pubsub.annotation.Topic
// end::imports[]

// tag::clazz[]
@PubSubClient // <1>
interface AnimalClient {

	@Topic("animals") // <2>
	fun send(data: ByteArray) // <3>

}
// end::clazz[]

File: test-suite-python/src/main/python/micronaut/gcp/pubsub/quickstart/AnimalClient.py
from abc import ABC, abstractmethod

# tag::imports[]
from micronaut.gcp.pubsub.annotation import PubSubClient, Topic
# end::imports[]


# tag::clazz[]
@PubSubClient  # <1>
class AnimalClient(ABC):

    @Topic("animals")  # <2>
    @abstractmethod
    def send(self, data: bytes) -> None:  # <3>
        ...
# end::clazz[]
